interp/notation.mli

```diff
@@ -323,6 +325,48 @@ val declare_custom_entry_has_ident : string -> int -> unit
 val entry_has_global : notation_entry_level -> bool
 val entry_has_ident : notation_entry_level -> bool
 
+(** Entry keys for constr notations *)
+
+type side = Left | Right
+
+type production_position =
+  | BorderProd of side * Gramlib.Gramext.g_assoc option
```

Comment:
make doesn't enforce separation between directories (so you can have a dependency chain like a/a.ml -> b/b.ml -> a/c.ml) but dune does, so you need to add gramlib to interp/dune to do this (currently it's added to the dependencies of parsing/ not interp/)

Reply:
(or move this part of notation.ml to a parsing/ file if it doesn't need to be in interp/)